### PR TITLE
Update HAGS detection

### PR DESCRIPTION
**Pull request by prgmitchell**
### Description

Report that HAGS is enabled when the registry key isn't set.
### Motivation and Context

HAGS being enabled continues to result in OBS performance issues and if the registry key isn't set, it is probably on so this should be reported by the analyzer.
### How Has This Been Tested?

Ran the log analyzer code locally with my changes, tested logs containing the "On", "Probably On", and "Off" strings.

Logs:
https://obsproject.com/logs/u5zg2liWUO8GIVSd
https://obsproject.com/logs/wGReCZRmeiCWDoKH

### Types of changes

- Tweak (non-breaking change to improve existing functionality)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

**Comment by flaeri** [comment]
I've been thinking it would be worthwhile to expand this a bit to check for Nvidia GPUs. If this gets merged, then we're also going to be potentially labeling every AMD GPU with a "critical" flag, even tho it is false (no HAGS for amd).

Example of win11 AMD GPU (flagging critical HAGS):
<https://obsproject.com/logs/jIqo9fN2EliHq90w>

I'm not sure about Intel, and whether it has that tech, and if they do, is it problematic?

Imo, the safest bet if we wanted to have this check, would be to check for whether OBS is running on the Nvidia GPU (or at the very least if there is an Nvidia GPU in the system), and only then pop this check.

**Comment by prgmitchell** [comment]
@flaeri I agree and hadn't considered this yet. I will try to update to PR soon to only flag on NVIDIA GPU's.

**Comment by flaeri** [comment]
all good :) To be fair, even without this PR, we still manage to occasionally tag AMD system with the HAGS critical hehe.

**Comment by prgmitchell** [comment]
I have updated the PR to only trigger based on the "probably on" string if OBS is running on an NVIDIA GPU.

File: checks/windows.py [hunk 1]
```diff
@@ -210,9 +210,14 @@ def checkGameMode(lines):
 
 
 def checkWin10Hags(lines):
+    d3dAdapter = search('Loading up D3D11', lines)
+    hagsMessage = """The new Windows Hardware-accelerated GPU scheduling ("HAGS") added with Windows 10 version 2004 is currently known to cause performance and capture issues with OBS, games and overlay tools. It's a new and experimental feature and we recommend disabling it via <a href="ms-settings:display-advancedgraphics">this screen</a> or <a href="https://obsproject.com/wiki/How-to-disable-Windows-10-Hardware-GPU-Scheduler">these instructions</a>."""
```
Comment (WizardCM) [inline checks/windows.py:214]: I'd rather not specify which version of Windows 10 doesn't work, as I'm quite sure 22H2 doesn't work either. Let's just specify WIndows 10 entirely.

**Comment by prgmitchell** [comment]
@WizardCM updated PR to remove the Windows 10 version number from the message.

File: checks/windows.py [hunk 2]
```diff
@@ -210,9 +210,14 @@ def checkGameMode(lines):
 
 
 def checkWin10Hags(lines):
+    d3dAdapter = search('Loading up D3D11', lines)
+    hagsMessage = """The new Windows Hardware-accelerated GPU scheduling ("HAGS") added with Windows 10 is currently known to cause performance and capture issues with OBS, games and overlay tools. It's a new and experimental feature and we recommend disabling it via <a href="ms-settings:display-advancedgraphics">this screen</a> or <a href="https://obsproject.com/wiki/How-to-disable-Windows-10-Hardware-GPU-Scheduler">these instructions</a>."""
```
Comment (RytoEX) [inline checks/windows.py:214]: At this point, HAGS is nearly three years old, so we might as well stop calling it "new".

Comment (RytoEX) [inline checks/windows.py:214]:
```suggestion
    hagsMessage = """The Hardware-accelerated GPU scheduling ("HAGS") feature added with Windows 10 is currently known to cause performance and capture issues with OBS, games and overlay tools. It's an experimental feature and we recommend disabling it via <a href="ms-settings:display-advancedgraphics">this screen</a> or <a href="https://obsproject.com/wiki/How-to-disable-Windows-10-Hardware-GPU-Scheduler">these instructions</a>."""
```

**Review by RytoEX — CHANGES_REQUESTED** [review]

**Review by RytoEX — APPROVED** [review]
Looks good to me. Tested the logs provided and got the following results:

| Log | HAGS warning |
| ------------- | ------------- |
| https://obsproject.com/logs/u5zg2liWUO8GIVSd | Yes |
| https://obsproject.com/logs/wGReCZRmeiCWDoKH | Yes |
| https://obsproject.com/logs/jIqo9fN2EliHq90w | No |

Discussed off-thread with @prgmitchell and this seems to be the expected results.

**Pull request merged**